src/aap_eda/tasks/project.py

```diff
@@ -39,3 +39,40 @@ def sync_project(project_id: int):
     ProjectImportService().sync_project(project)
 
     logger.info(f"Task complete: Sync project ( project_id={project.id} )")
+
+
+@job("default")
+def monitor_project_tasks() -> None:
+    """Handle project tasks that are stuck.
+
+    Check if there are projects in PENDING state that doesn't have
+    any related job in the queue. If there are any, put them back
+    to the queue.
+    """
+    logger.info("Task started: Monitor project tasks")
+
+    queue = get_queue("default")
+
+    # Filter projects that doesn't have any related job
```

Comment:
@Alex-Izquierdo Instead of using the monitor can we use the timeout feature in RQ jobs and catch the exception JobTimeoutException and do the cleanup. When we enqueue a task we can specify a timeout value so that way we can specify the max time to wait for a project sync to finish. It would be nice if the ProjectSync UI screen allows the user to specify a timeout value, we would use the default defined in the settings file but override it in the UI.

Reply:
@mkanoor I agree with adding a default timeout but this is not the case with this fix. This is to handle incomplete jobs that have been lost due to a broker/worker restart.